packages/api/src/command/medical/patient/consolidated-get.ts

```diff
@@ -33,12 +37,58 @@ export async function getConsolidatedPatientData({
   resources?: ResourceTypeForConsolidation[];
   dateFrom?: string;
   dateTo?: string;
-}): Promise<Bundle<Resource>> {
-  const { log } = Util.out(`[getConsolidatedPatientData - cxId ${cxId}, patientId ${patientId}]`);
+}): Promise<QueryProgress> {
+  const { log } = Util.out(`queryDocumentsAcrossHIEs - M patient ${patientId}`);
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+  if (patient.data.consolidatedQuery?.status === "processing") {
+    log(`Patient ${patientId} consolidatedQuery is already 'processing', skipping...`);
+    return patient.data.consolidatedQuery;
+  }
+  const progress: QueryProgress = { status: "processing" };
+  await udpateConsolidatedQueryProgress({
+    patient,
+    progress,
+    reset: true,
+  });
+  getConsolidatedAndSendToCx({ patient, resources, dateFrom, dateTo }).catch(emptyFunction);
+  return progress;
+}
```

Comment:
Shouldn't we update [the docs](https://docs.metriport.com/medical-api/api-reference/fhir/get-patient-consolidated) to reflect this new async flow?

Reply:
Yeah, its marked as outstanding on the PR description, it will come on a downstream PR.